app.py

```diff
@@ -262,57 +257,45 @@ async def get_pins():
             pins = await cur.fetchall()
             await cur.close()
             pool.close()
-            return jsonify({'pins': pins}), 200
+            pins_dict_list = [dict(zip(("pin_id", "title", "body", "image", "user_id", "added_date"), row)) for row in
+                              pins]
+            return jsonify({"pins": pins_dict_list}), 200
 
 
 # Get Pin By ID
-@app.get('/pins/<int:pin_id>')
+@app.get("/pins/<int:pin_id>")
 async def get_pin(pin_id):
-    pin=await check_pin_exists2(pin_id)
+    pin = await check_pin_exists(pin_id=pin_id)
     if not pin:
-        return jsonify({'error': 'Pin does not exist'}), 404
-    pool = await create_pool()
-    async with pool.acquire() as conn:
-        async with conn.cursor() as cur:
-            await cur.execute("SELECT * FROM pins where pin_id =%s", (pin_id,))
-            pin = await cur.fetchone()
-            await cur.close()
-            pool.close()
-            return jsonify({'pin': pin}), 200
+        return jsonify({"error": "Pin does not exist"}), 404
+    pin_dict = dict(zip(("pin_id", "title", "body", "image", "user_id", "added_date"), pin))
+    return jsonify({"pin": pin_dict}), 200
 
 
 # Check if pin exists and return pin
-async def check_pin_exists(user_id, pin_id):
+async def check_pin_exists(user_id=None, pin_id=None):
     pool = await create_pool()
     async with pool.acquire() as conn:
         async with conn.cursor() as cur:
-            await cur.execute("SELECT * FROM pins where pin_id =%s and user_id =%s", (pin_id, user_id))
-            pin = await cur.fetchone()
-            await cur.close()
-            pool.close()
-            return pin
-
-
-async def check_pin_exists2(pin_id):
-    pool = await create_pool()
-    async with pool.acquire() as conn:
-        async with conn.cursor() as cur:
-            await cur.execute("SELECT * FROM pins where pin_id =%s", (pin_id))
+            if not user_id:
+                await cur.execute("SELECT * FROM pins where pin_id =%s", (pin_id))
+            else:
+                await cur.execute("SELECT * FROM pins where pin_id =%s and user_id =%s", (pin_id, user_id))
             pin = await cur.fetchone()
             await cur.close()
             pool.close()
             return pin
 
 
 # Update pin
-@app.put('/pins/<int:pin_id>')
+@app.put("/pins/<int:pin_id>")
 @token_required
 async def update_pin(current_user, pin_id):
```

Comment:
With this implementation, anyone could modify anyone's pins which is not good. So make sure that only the author can modify it.

Reply:
I handled this scenario. check_pin_exists method checked the login user_id and pin_id these are match or not. Therefor author can  delete or update only their pins